server/src/Api/Program.cs

```diff
@@ -12,6 +14,8 @@
     .AddInfrastructure(builder.Configuration)
     .AddApi();
 
+AspNetCoreResult.Setup(config => config.DefaultProfile = new CustomAspNetCoreResultEndpointProfile());
```

Comment:
We can register the profile as a scoped service so we can use dependency injection and access our stuff in the profile.
```suggestion
services.AddHttpContextAccessor();
services.AddScoped<IAspNetCoreResultEndpointProfile, ResultEndpointProfile>();
```